develop/docs/tracing.md

```diff
@@ -100,3 +102,106 @@ global `TracerProvider`.
+any reduction in that noise will be a good idea. Not to mention the consistency
+benefit of sharing a single mapping function.
+
+### Start a span in `common` or other non-service-specific code
+
+*Q:* Given that common code can be called from any service, how can I start a span
+in common library code that is bound the the appropriate service
+(frontend/history/matching/worker)?
+
+*A:* The `TracerProvider` that created the currently active Span can be retrieved
+from that Span itself and the currently active Span can be received from from
+the `context.Context`.
+
+```
+// DoFoo is a function in the common package
+func DoFoo(ctx context.Context, x int, y string) string {
+   var span trace.Span
+   ctx, span = trace.SpanFromContext(ctx).TracerProvider().Tracer("go.temporal.io/server/common").Start("DoFoo")
+   defer span.End()
+   return fmt.Sprintf("%v-%v", y, x)
+}
+```
+
+### `RecordError` does not imply Span failure
+
+Using `Span.RecordError` is a good idea but not all errors imply failure. Thus
+if you want to capture an error _and also_ capture that a spain failed, you must
+additionally call `Span.SetStatus(codes.Error, err.Error())`. A
+`FailSpanWithError` utility function might be a good idea.
+
+### Propagate TraceContext across things other than function calls
+
+This is taken care of by default for gRPC calls via the otelgrpc interceptors.
+However you may want to propagate tracing information between goroutines or
+other places where the `context.Context` is not passed such as handoffs through
+a Go channel or an external datastore. There are two broad approaches that are
+applicable in different situations:
+
+1. If the object being transferred is not externally durable (e.g. an object put
+   into a Go channel but _not_ spooled to a dataabse) then you can pull the
+   `trace.SpanContext` out of the current `trace.Span` with
+   `trace.SpanContextFromContext(context.Context)` or `Span.SpanContext()` and
+   pass that object along with the data being transferred. The consuming side
+   can restore the tracing state with
+   `trace.ContextWithSpanContext(trace.SpanContext)`.
+1. If the tracing state needs to be serialized, the OTEL library provides the
+   [propagation](https://pkg.go.dev/go.opentelemetry.io/otel@v1.7.0/propagation)
+   package to convert trace state into a more serialization-friendly type such
+   as a `map[string]string`. The `propagation.TraceContext` type can be used to
+   inject and extract trace state into a key-value-ish object.
+
+```
+carrier := propagation.MapCarrier(map[string]string{})
+propagation.TraceContext{}.Inject(ctx, carrier)
+// write the carrier object to a durable store
+```
+### Trace individual tasks that are processed together in batches
+
+OpenTelemetry Spans can be _linked_ together to form a non-parent-child
+relationship. One of the main use cases for linking is so that a batch process
+(e.g. a database read that fills a large buffer of work itemsindividual tasks
```

Comment:
typo: itemsindividual

Reply:
actually delete this whole ¶ as a dupe?

Reply:
Lol. I distinctly remember fat-fingering in vim and then I couldn't figure out what changed :disappointed: